### PR TITLE
Auto limit and normalize skin weights

### PR DESCRIPTION
**Pull request by pjoe**
Limit to 4 joints and normalize when 'all vertex influences' export setting is off

### CODE REVIEW

**Comment by pjoe** [comment]
CI: https://circleci.com/gh/p3d-in/glTF-Blender-IO/31

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py [hunk 1]
```diff
@@ -165,10 +165,10 @@ def __gather_skins(blender_primitive, export_settings):
         joint_id = 'JOINTS_' + str(bone_index)
         weight_id = 'WEIGHTS_' + str(bone_index)
         while blender_primitive["attributes"].get(joint_id) and blender_primitive["attributes"].get(weight_id):
-            if bone_index >= 4:
-                gltf2_io_debug.print_console("WARNING", "There are more than 4 joint vertex influences."
-                                                        "Consider to apply blenders Limit Total function.")
+            if bone_index >= 1:
```
Comment (donmccurdy) [inline addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py]: Why `>= 1` here?

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py [hunk 2]
```diff
@@ -192,6 +192,15 @@ def __gather_skins(blender_primitive, export_settings):
 
             # weights
             internal_weight = blender_primitive["attributes"][weight_id]
+            # normalize first 4 weights, when not exporting all influences
+            if not export_settings['gltf_all_vertex_influences']:
+                for idx in range(0, len(internal_weight), 4):
```
Comment (donmccurdy) [inline addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py]: This is normalizing weights 1-4, 5-8, 9-12, each group in batches of four, correct? That seems a little confusing, doesn't look like we should need this outer loop.

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py (hunk 1 above)
Comment (pjoe) [reply addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py]: Heh, was pretty confused about why the 4 wasn't working, however bone_index here refers to the output groups in gltf (JOINTS_X, WEIGHTS_X) which each hold 4 bones.

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py (hunk 2 above)
Comment (pjoe) [reply addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py]: Yep, it needs to normalize each bundle of 4. Not sure how you would do that without the outer loop in line 197

Comment (donmccurdy) [reply addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py]: Oh I see, `internal_weights` is the weights for all vertices, and each "bundle of 4" is one vertex? I think I'd misread and assumed this was normalizing WEIGHTS_0, then normalizing WEIGHTS_1, etc.

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py (hunk 1 above)
Comment (donmccurdy) [reply addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitive_attributes.py]: Oh that's confusing. 😅 Do you mind renaming this variable to `group_index` or `set_index`?

**Comment by emackey** [comment]
@donmccurdy I haven't dug much into bones & skinning, so this one is for you & Norbert.

**Comment by donmccurdy** [comment]
This is working correctly for me. Thanks @pjoe!

**Pull request merged**